test/integration/sessions_test.rb

```diff
@@ -1,10 +1,22 @@
 require "integration_test_helper"
+require "gds_api/test_helpers/account_api"
 
 class SessionsTest < ActionDispatch::IntegrationTest
+  include GdsApi::TestHelpers::AccountApi
+
   context "when logged in" do
-    should "redirect the user to the account manager URL if they visit /sign-up" do
+    should "redirect the user to the account manager URL if they visit /sign-in" do
       get "/sign-in", headers: { "GOVUK-Account-Session" => "placeholder" }
       assert_redirected_to Plek.find("account-manager")
     end
   end
+
+  context "when logged out" do
+    should "add a redirect_path param to /sign-in from the HTTP REFERRER header" do
+      stub_account_api_get_sign_in_url(redirect_path: "/bank-holiday")
+
+      get "/sign-in", headers: { "HTTP_REFERER" => "#{Plek.new.website_root}/bank-holiday" }
```

Comment:
The request header to send in the test is `Referer`, not `HTTP_REFERER`, isn't it?

Reply:
It would be good to have a couple more tests:

1. Set both a `redirect_path` and a `Referer` - check that the `redirect_path` wins
2. Set a `Referer` going to some other domain - check that it's not used

To be very explicit, the test could even assert that the stub has been called (making it more likely to fail if a bug gets introduced later):

```ruby
stub = stub_account_api_get_sign_in_url(redirect_path: "/from-referer")

get "/sign-in", headers: { Referer: "#{Plek.new.website_root}/from-referer" }

assert_response :redirect
assert_requested stub
```

```ruby
stub_params = stub_account_api_get_sign_in_url(redirect_path: "/from-params")
stub_referer = stub_account_api_get_sign_in_url(redirect_path: "/from-referer")

get "/sign-in", headers: { Referer: "#{Plek.new.website_root}/from-referer" }, params: { redirect_path: "/from-params" }

assert_response :redirect
assert_requested stub_params
assert_not_requested stub_referer
```


```ruby
stub_account_api_get_sign_in_url
stub = stub_account_api_get_sign_in_url(redirect_path: "/from-referer")

get "/sign-in", headers: { Referer: "//evil/from-referer" }

assert_response :redirect
assert_not_requested stub
```

Reply:
you are right using `/from-referer` is much clearer than the bank holiday path.
amended those.

I like the evil path test too, added that.
Just checking on the Referer header... These name changes within rails are still confusing me I think.

Reply:
Thanks these are amended